library/src-bootstrapped/scala/StagedTuple.scala

```diff
@@ -203,7 +203,7 @@ object StagedTuple {
     if (!specialize) '{dynamic_++[Self, That]($self, $that)}
     else {
       def genericConcat(xs: Expr[Tuple], ys: Expr[Tuple]): Expr[Tuple] =
-        fromArrayStaged[Tuple]('{${ toArrayStaged(xs, None) } ++ ${ toArrayStaged(ys, None) }}, None)
+        fromArrayStaged[Tuple]('{${ toArrayStaged(xs, None) } ++ (${ toArrayStaged(ys, None) }: Array[Object])}, None)
```

Comment:
Why is the ascription needed here?

Reply:
From the desugaring of `${...}` to `def exprSplice[T](x: Expr[T]): T` we get now need to infer `T` as well as the types of `++`. 

The acctual error is
```scala
[error] 206 |        fromArrayStaged[Tuple]('{${ toArrayStaged(xs, None) } ++ ${ toArrayStaged(ys, None) }}, None)
[error]     |                                                                    ^^^^^^^^^^^^^^^^^^^^^^^
[error]     |    Found:    quoted.Expr[Array[Object]]
[error]     |    Required: quoted.Expr[scala.collection.GenTraversableOnce[Object]]
```

Reply:
That's problematic. We should avoid type ascriptions that need to be added for obscure reasons.  Should we change the desugaring instead?

Reply:
Actually, this is an instance of #6126 which is quite specific to `++` with `Array`s. If fixed this would not be an issue.

Reply:
I'll have a look at #6126.